starknet_devnet/postman_wrapper.py

```diff
@@ -0,0 +1,54 @@
+"""
+This module wraps the usage of Postman for L1 <> L2 interaction.
+"""
+from abc import ABC, abstractmethod
+from web3 import HTTPProvider, Web3
+from starkware.contracts.utils import load_nearby_contract
+
+from .postman.postman import Postman
+from .postman.eth_test_utils import EthAccount, EthContract
+
+from .constants import TIMEOUT_FOR_WEB3_REQUESTS
+
+
+class PostmanWrapper(ABC):
+    """Postman Wrapper base class"""
+
+    @abstractmethod
+    def __init__(self):
+        self.postman: Postman = None
+        self.web3: Web3 = None
+        self.mock_starknet_messaging_contract: EthContract = None
+        self.eth_account: EthAccount = None
+
+    @abstractmethod
+    def get_mock_messaging_contract_in_l1(self, starknet, contract_address):
+        """Retrieves the Mock Messaging contract deployed in an L1 network"""
+
+    async def flush(self):
+        """Handles the L1 <> L2 message exchange"""
+        await self.postman.flush()
+
+class GanachePostmanWrapper(PostmanWrapper):
+    """Wrapper of Postman usage on a local testnet instantiated using Ganache"""
+
+    def __init__(self, network_url: str):
+        super().__init__()
+        request_kwargs = {"timeout": TIMEOUT_FOR_WEB3_REQUESTS}
+        self.web3 = Web3(HTTPProvider(network_url, request_kwargs=request_kwargs))
+        self.eth_account = EthAccount(self.web3,self.web3.eth.accounts[0])
+
+    def __deploy(self, starknet):
+        self.mock_starknet_messaging_contract = self.eth_account.deploy(load_nearby_contract("MockStarknetMessaging"))
+        self.postman = Postman(self.mock_starknet_messaging_contract,starknet)
```

Comment:
This line is already executed in get_mock_messaging_contract_in_l1, it can just be unindented there and removed from here.